### PR TITLE
Fix the `shorten-64-to-32` warnings

### PR DESCRIPTION
**Pull request by neonichu**
- Disable warnings for any target using LLVM sources
- Fix warnings in the actual llbuild codebase

### CODE REVIEW

File: include/llbuild/Basic/ShellUtility.h [hunk 1]
```diff
@@ -13,8 +13,11 @@
 #ifndef LLBUILD_BASIC_SHELLUTILITY_H
 #define LLBUILD_BASIC_SHELLUTILITY_H
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshorten-64-to-32"
```
Comment (ddunbar) [inline include/llbuild/Basic/ShellUtility.h]: I would rather not do this, it is too hard to maintain. I don't find this warning really adds enough value to be worth the impact on the code, I would rather turn it off TBH.

Comment (neonichu) [inline include/llbuild/Basic/ShellUtility.h]: OK, I reverted that and instead disabled the warning for most targets.

File: perftests/Xcode/PerfTests/CorePerfTests.mm [hunk 2]
```diff
@@ -63,7 +63,7 @@ static int32_t IntFromValue(const core::ValueType& Value) {
 
   virtual void start(BuildEngine& Engine) override {
     // Request all of the inputs.
-    for (int i = 0, e = Inputs.size(); i != e; ++i) {
+    for (size_t i = 0, e = Inputs.size(); i != e; ++i) {
```
Comment (ddunbar) [inline perftests/Xcode/PerfTests/CorePerfTests.mm]: `auto` is maybe the best choice for places where we are inheriting the type from `.size()`.

File: lib/Commands/NinjaBuildCommand.cpp [hunk 3]
```diff
@@ -146,7 +146,7 @@ struct BuildExecutionQueue {
   bool useLIFO;
 
 public:
-  BuildExecutionQueue(unsigned numLanes, bool useLIFO)
+  BuildExecutionQueue(unsigned long numLanes, bool useLIFO)
```
Comment (ddunbar) [inline lib/Commands/NinjaBuildCommand.cpp]: I would rather not use `long` in interfaces... I think it is rarely the right type. If a larger type than int is needed then `uint64_t` is my preferred type (which is not platform dependent). In this case, though, the extra size is never needed here.

Comment (ddunbar) [inline lib/Commands/NinjaBuildCommand.cpp]: ... actually, that is generally my feeling about _most_ of the changes in this PR, I would rather avoid long (and cast it away, when necessary), or use `uint64_t` in the few cases where we need the precision.

**Pull request closed without merging**